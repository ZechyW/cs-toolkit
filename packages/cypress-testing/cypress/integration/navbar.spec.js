describe("The main app navbar", () => {
  beforeEach(() => {
    cy.visit("/");
  });

  it("shows the burger menu when the width is less than 1088px", () => {
    cy.viewport(1088, 500);
    cy.get(".navbar-burger").should("not.be.visible");
    cy.viewport(1087, 500);
    cy.get(".navbar-burger").should("be.visible");
  });

  describe("toggles", () => {
    it("on click", () => {
      cy.get(".navbar")
        .should("have.class", "is-expanded")
        .click()
        .should("not.have.class", "is-expanded")
        .click()
        .should("have.class", "is-expanded");
    });

    it("only if the burger menu is collapsed", () => {
      cy.viewport(1087, 500);
      cy.get(".navbar-burger")
        .click()
        .should("have.class", "is-active");

      cy.get(".navbar")
        .should("not.have.class", "is-expanded")
        .click()
        .should("not.have.class", "is-expanded");
    });
  });

  it("collapses when window is scrolled", () => {
    cy.scrollTo(0, 1);
    cy.get(".navbar").should("not.have.class", "is-expanded");
  });

  it("Snapshot - Expanded", () => {
    cy.get(".navbar")
      .should("have.class", "is-expanded")
      .and("not.have.class", "is-animating");
    cy.get(".navbar").toMatchSnapshot();
  });

  it("Snapshot - Collapsed", () => {
    // Should collapse on click
    cy.get(".navbar")
      .click()
      .should("not.have.class", "is-expanded")
      .and("not.have.class", "is-animating");
    cy.get(".navbar").toMatchSnapshot();
  });
});
