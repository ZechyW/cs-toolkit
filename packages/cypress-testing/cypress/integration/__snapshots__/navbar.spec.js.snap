exports[`The main app navbar > Snapshot - Expanded #0`] = `
<nav
  class="navbar is-light is-fixed-top is-expanded"
  style="height: 10rem; padding: 3rem 1.5rem;"
>
  <div class="navbar-brand">
    <div
      class="navbar-item has-cursor-pointer"
      style="padding: 0rem; margin-left: 0px; margin-right: 0px;"
    >
      <img
        src="/static/media/logo.faf9ea68.png"
        alt="CS Toolkit"
        style="height: 4rem; width: 4rem; margin-right: 1rem; max-height: 100%;"
      />
      <div>
        <p class="title">The Code Switching Toolkit</p>
        <p class="subtitle" style="height: 1.6rem; opacity: 1;">
          A (roughly) Minimalist framework for exploring code switching data
        </p>
      </div>
    </div>
    <div
      role="button"
      class="navbar-burger"
      aria-label="menu"
      aria-expanded="false"
    >
      <span aria-hidden="true"></span><span aria-hidden="true"></span
      ><span aria-hidden="true"></span>
    </div>
  </div>
  <div class="navbar-menu">
    <div class="navbar-start"></div>
    <div class="navbar-end">
      <div class="navbar-item">
        <div class="buttons">
          <button class="button is-primary">
            <strong>Reset Layout</strong>
          </button>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`The main app navbar > Snapshot - Collapsed #0`] = `
<nav
  class="navbar is-light is-fixed-top"
  style="height: 3.25rem; padding: 0rem;"
>
  <div class="navbar-brand">
    <div
      class="navbar-item has-cursor-pointer"
      style="padding: 0.5rem 0.75rem; margin-left: 0px; margin-right: 0px;"
    >
      <img
        src="/static/media/logo.faf9ea68.png"
        alt="CS Toolkit"
        style="height: 1.75rem; width: 1.75rem; margin-right: 0.5rem; max-height: 100%;"
      />
      <div>
        <p class="title">CS Toolkit</p>
        <p class="subtitle" style="height: 0rem; opacity: 0;">
          A (roughly) Minimalist framework for exploring code switching data
        </p>
      </div>
    </div>
    <div
      role="button"
      class="navbar-burger"
      aria-label="menu"
      aria-expanded="false"
    >
      <span aria-hidden="true"></span><span aria-hidden="true"></span
      ><span aria-hidden="true"></span>
    </div>
  </div>
  <div class="navbar-menu">
    <div class="navbar-start"></div>
    <div class="navbar-end">
      <div class="navbar-item">
        <div class="buttons">
          <button class="button is-primary">
            <strong>Reset Layout</strong>
          </button>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
